### ISSUE TITLE
wildfly15/jdk11 potential issue in future?

### ISSUE THREAD

**Issue by sschmiedleitner** [elastic/apm-agent-java#395]
Today I tried the latest agent version 1.2.0 together with wildfly15/jdk11. To run it we use docker image `jboss/wildfly:15.0.0.Final-1` as base.

On startup some warnings are shown in the logs you probably should take a look at:

```
=========================================================================

  JBoss Bootstrap Environment

  JBOSS_HOME: /opt/jboss/wildfly

  JAVA: /usr/lib/jvm/java/bin/java

  JAVA_OPTS:  -server -Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true -javaagent:/opt/jboss/wildfly/elastic-apm-agent-1.2.0.jar  --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-modules=java.se

=========================================================================

OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
2018-12-20 12:39:20.695 [apm-server-healthcheck] INFO co.elastic.apm.agent.report.ApmServerHealthChecker - Elastic APM server is available: {"ok":{"build_date":"2018-11-09T17:48:30Z","build_sha":"2c46a2919dbe06767caeacbffd0055aab1493991","version":"6.5.0"}}
2018-12-20 12:39:20.732 [main] INFO co.elastic.apm.agent.configuration.StartupInfo - Starting Elastic APM 1.2.0 as pst on Java 11.0.1 (Oracle Corporation) Linux 4.4.0-138-generic
2018-12-20 12:39:20.733 [main] WARN co.elastic.apm.agent.configuration.StartupInfo - To enable all features and to increase startup times, please configure application_packages
2018-12-20 12:39:21,891 INFO  [org.jboss.modules] (main) JBoss Modules version 1.8.7.Final
...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by co.elastic.apm.agent.shaded.bytebuddy.dynamic.loading.ByteArrayClassLoader$SynchronizationStrategy$ForJava7CapableVm to method java.lang.ClassLoader.getClassLoadingLock(java.lang.String)
WARNING: Please consider reporting this to the maintainers of co.elastic.apm.agent.shaded.bytebuddy.dynamic.loading.ByteArrayClassLoader$SynchronizationStrategy$ForJava7CapableVm
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
...
```

could you please consider to check this?

**Comment by eyalkoren** [comment]
@sschmiedleitner Thanks for reporting! Is it working otherwise? Any other symptoms? Are you using security manager?

**Comment by sschmiedleitner** [comment]
in general it is working, so it is transmitting data to apm-server/elasticsearch and I can see results in the APM UI. But just to be prepared for the future, I reported this.

**Comment by eyalkoren** [comment]
The warnings above are since Java 9 modules were introduced- they warn about code making accesses that break the module dependency setup. Currently, the default value used for the `--illegal-access` flag is `permit`, which emits a warning during the first fault access, but [this default will be changed in future releases](http://mail.openjdk.java.net/pipermail/jigsaw-dev/2017-June/012841.html).

I tried to reproduce but I got other warnings from WildFly classes and even when used `--illegal-access=debug` couldn't reproduce, however- seems like the wrong `SynchronizationStrategy` is chosen in Java 11, for which I opened an issue in ByteBuddy repo - https://github.com/raphw/byte-buddy/issues/603.

Since everything seems to be working properly, apart from these warnings, I am closing this issue and will follow the ByteBuddy issue.
